From fa7d88a54d572fe1e5cd8fe705070c889dc50da6 Mon Sep 17 00:00:00 2001
From: Yijie Yang <yijie.yang@oss.qualcomm.com>
Date: Thu, 15 Jan 2026 10:30:10 +0800
Subject: [PATCH] QCLINUX: arm64: dts: qcom: workaround for TPM pins

Purwa will crash if HLOS attempts to access or configure the TPM pins, as
these operations require higher privilege levels. Therefore, the TPM pins
should be reserved in the DTS until NHLOS provides proper support.

Upstream-Status: Pending
Signed-off-by: Yijie Yang <yijie.yang@oss.qualcomm.com>
---
 arch/arm64/boot/dts/qcom/purwa-iot-som.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/purwa-iot-som.dtsi b/arch/arm64/boot/dts/qcom/purwa-iot-som.dtsi
index 71770a8504cf..238e38b5dd73 100644
--- a/arch/arm64/boot/dts/qcom/purwa-iot-som.dtsi
+++ b/arch/arm64/boot/dts/qcom/purwa-iot-som.dtsi
@@ -483,7 +483,8 @@ &remoteproc_cdsp {
 };
 
 &tlmm {
-	gpio-reserved-ranges = <34 2>; /* TPM LP & INT */
+	gpio-reserved-ranges = <34 2>, /* TPM LP & INT */
+			       <44 4>; /* SPI (TPM) */
 
 	pcie3_default: pcie3-default-state {
 		clkreq-n-pins {
